### PR TITLE
Add month_length, quarter_length and year_length methods

### PR DESCRIPTION
**Pull request by danstewart**
Half of #66

### CODE REVIEW

**Comment by danstewart** [comment]
Yeah - fixing now, sorry.

**Comment by danstewart** [comment]
@preaction I've squashed another commit in (hopefully) fixing the issues. Travis is still running the tests for the first commit so might be a while. Thanks.

**Comment by preaction** [comment]
Looks like there's still a tidiness problem. [Here's how to run `tidyall` to fix the problem](https://github.com/houseabsolute/DateTime.pm/blob/master/CONTRIBUTING.md#tidyall)

**Comment by danstewart** [comment]
Weird, I did run `tidyall -a`
I'll have a look at it tomorrow, away from my computer now.

**Comment by danstewart** [comment]
Ah, I think the issue is with a space before the bracket on the tests - I might have ran `tidyall` in `/lib`

File: t/03components.t [hunk 1]
```diff
@@ -83,6 +83,10 @@ use DateTime;
     ok( !$d->is_leap_year,         '->is_leap_year' );
     ok( !$d->is_last_day_of_month, '->is_last_day_of_month' );
 
+    is ( $d->month_length,   31,  '->month_length' );
+    is ( $d->quarter_length, 92,  '->quarter_length' );
+    is ( $d->year_length,    365, '->year_length' );
+
```
Comment (autarch) [inline t/03components.t]: This needs tests for leap years too, and ideally for every quarter in both leap & non-leap years.

**Comment by autarch** [comment]
Re: tidying - the latest perltidy release changed some tidying for our config, so I pushed an update to master. No branches based off the old master will pass the tidy test. You can rebase off master to get the fixes.

**Comment by autarch** [comment]
Merged from the CLI. Thanks!

**Pull request closed without merging**